service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -115,6 +115,60 @@ private String getResourceLocation(Workspace workspace, String requestedLocation
         : requestedLocation;
   }
 
+  private String getResourceZone(Workspace workspace, String requestedZone) {
+    if (Strings.isNullOrEmpty(requestedZone)) {
+      String location =
+          workspace
+              .getProperties()
+              .getOrDefault(
+                  WorkspaceConstants.Properties.DEFAULT_RESOURCE_LOCATION,
+                  GcpResourceConstant.DEFAULT_REGION);
+      switch (location) {
+        case "asia-east1":
```

Comment:
do we even support all these regions?

Reply:
https://cs.github.com/verily-src/terra-service-resource-buffer/blob/2fbba28d70e761ad56564ff8a48f2180e080cbe9/deployment/resource-buffer/configs/prod/resource-config/resource_verilyprod_v4.yml#L42
looks like these are blocked?

makes me nervous that we are not reading from the source. random thought: there is this https://cloud.google.com/compute/docs/reference/rest/v1/regions/list api that we can use to validate if the user's regional preference is even feasible.